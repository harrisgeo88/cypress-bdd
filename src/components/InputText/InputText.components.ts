import styled from 'styled-components'

export const InputTextWrapper = styled.div`
  margin: 10px 0;
  height: 50px;
`

export const Input = styled.input`
  padding: 0 10px;
  height: 100%;
  width: calc(100% - 20px);
  border: 0;
  border-radius: 5px;
  border: 1px solid grey;
  font-size: 14px;
`
