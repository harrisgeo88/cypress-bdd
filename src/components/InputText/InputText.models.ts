// interface HTMLInput extends React.HTMLProps<HTMLInputElement> {}

type CustomHTMLInput = Omit<React.HTMLProps<HTMLInputElement>, 'onChange'>

export interface InputTextProps extends CustomHTMLInput {
  value: string
  onChange: (value: string) => void
}
