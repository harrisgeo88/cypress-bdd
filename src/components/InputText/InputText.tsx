import React from 'react'
import { InputTextWrapper, Input } from './InputText.components'
import { InputTextProps } from './InputText.models'

export const InputText = ({ onChange, ...rest }: InputTextProps) => {
  const handleChange = (e) => {
    const target = e.target as HTMLInputElement
    onChange && onChange(target.value)
  }

  return (
    <InputTextWrapper>
      <Input {...rest} onChange={handleChange} />
    </InputTextWrapper>
  )
}
