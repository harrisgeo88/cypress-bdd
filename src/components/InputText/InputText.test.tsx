import { InputText } from './InputText'
import { render, fireEvent } from '@testing-library/react'

describe('given an `InputText` component', () => {
  describe('when it is rendered', () => {
    it('should match snapshot', () => {
      const { container } = render(<InputText value="" onChange={jest.fn()} />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('when its text changes', () => {
    it('should return its value with `onChange`', () => {
      let inputValue = ''
      const props = {
        value: '',
        'data-testid': 'input',
        onChange: (value) => {
          inputValue = value
        },
      }
      const { rerender, getByTestId } = render(<InputText {...props} />)
      const Input = getByTestId('input')
      fireEvent.change(Input, {
        target: { value: 'some input' },
      })
      rerender(<InputText {...props} value="some input" />)
      expect(inputValue).toEqual('some input')
    })
  })
})
