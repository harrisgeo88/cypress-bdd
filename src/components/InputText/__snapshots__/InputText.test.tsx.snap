// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given an \`InputText\` component when it is rendered should match snapshot 1`] = `
<div
  class="InputTextcomponents__InputTextWrapper-sc-11nm3a5-0 ehojjc"
>
  <input
    class="InputTextcomponents__Input-sc-11nm3a5-1 lhFXMG"
    value=""
  />
</div>
`;
